cmd/observe/io_reader_observer.go

```diff
@@ -87,7 +88,24 @@ func (c *ioReaderClient) Recv() (*observer.GetFlowsResponse, error) {
 	for c.scanner.Scan() {
 		line := c.scanner.Text()
 		var res observer.GetFlowsResponse
-		err := protojson.Unmarshal(c.scanner.Bytes(), &res)
+		err := protojson.UnmarshalOptions{DiscardUnknown: c.discardUnknown}.Unmarshal(c.scanner.Bytes(), &res)
+		if err != nil && !c.discardUnknown {
+			prevErr := err
+			// the error might be that the JSON data contains an unknown field.
+			// This can happen we attempting to decode flows generated from a
+			// newer Hubble version than the CLI (having introduced a new
+			// field). Retry parsing discarding unknown fields and see whether
+			// the decoding is successful.
```

Comment:
is there nothing in the error that we can check to make sure it was unknown field, rather than attempting to parse twice?

Reply:
Tracking the error from protobuf it's from https://github.com/cilium/hubble/blob/bb32734273c5f7a90d38dfde0f709867e06518f9/vendor/google.golang.org/protobuf/encoding/protojson/decode.go#L197 and https://github.com/cilium/hubble/blob/bb32734273c5f7a90d38dfde0f709867e06518f9/vendor/google.golang.org/protobuf/encoding/protojson/decode.go#L96-L101

So I see no way to `errors.Is` as it isn't wrapped. Matching on the error message doesn't feel right either.